daemon/api.go

```diff
@@ -2635,56 +2627,13 @@ func postApps(c *Command, r *http.Request, user *auth.UserState) Response {
 		return InternalError("no services found")
 	}
 
-	// the argv to call systemctl will need at most one entry per appInfo,
-	// plus one for "systemctl", one for the action, and sometimes one for
-	// an option. That's a maximum of 3+len(appInfos).
-	argv := make([]string, 2, 3+len(appInfos))
-	argv[0] = "systemctl"
-
-	argv[1] = inst.Action
-	switch inst.Action {
-	case "start":
-		if inst.Enable {
-			argv[1] = "enable"
-			argv = append(argv, "--now")
-		}
-	case "stop":
-		if inst.Disable {
-			argv[1] = "disable"
-			argv = append(argv, "--now")
-		}
-	case "restart":
-		if inst.Reload {
-			argv[1] = "reload-or-restart"
-		}
-	default:
-		return BadRequest("unknown action %q", inst.Action)
-	}
-
-	snapNames := make([]string, 0, len(appInfos))
-	lastName := ""
-	names := make([]string, len(appInfos))
-	for i, svc := range appInfos {
-		argv = append(argv, svc.ServiceName())
-		snapName := svc.Snap.Name()
-		names[i] = snapName + "." + svc.Name
-		if snapName != lastName {
-			snapNames = append(snapNames, snapName)
-			lastName = snapName
+	chg, err := servicectl.ServiceControl(st, appInfos, &inst)
```

Comment:
Please feel free to postpone to Change to TaskSet modification. The reason why we do this, for the record, is because task sets are composable. We can have those same tasks next to other tasks.

Reply:
Allright, postponing this for now. Yes, I'm aware of the distinction between tasksets and changes, it was just more handy to have the change right away for the two use cases we currently have.

Reply:
Can you please do the rename suggested, though?  There's no reason for this to be completely different, per prior comment.

Reply:
hmm... hopping on late, but it seems to me that getting something back from a function called `Change`, and then having to pass it to a different function called `NewChange` is going to be a little bit weird for casual code readers.

Reply:
@chipaca For now though this name makes sense. If we later change it to return taskset then indeed it might make sense to rename it.